lib/elixir/lib/enum.ex

```diff
@@ -2259,14 +2266,12 @@ defmodule Enum do
     do: []
   def take_random(first..first, count) when is_integer(count) and count >= 1,
     do: [first]
-  def take_random(first..last, 1) when first > last,
-    do: take_random(last..first, 1)
-  def take_random(first..last, 1),
-    do: [random_index(last - first) + first]
+  def take_random(_.._ = range, 1),
+    do: [random(range)]
```

Comment:
This changes the semantic of the code, doesn't it? It means that, the following will no longer be true when starting from the same seed:

```elixir
:rand.seed(:exsplus, {101, 102, 103})
one = Enum.take_random(1..100, 1)

:rand.seed(:exsplus, {101, 102, 103})
two = Enum.take_random(1..100, 2)

hd(one) == hd(two)
```